packages/modules/mssqlserver/src/mssqlserver-container.ts

```diff
@@ -0,0 +1,123 @@
+import { AbstractStartedContainer, GenericContainer, StartedTestContainer } from "testcontainers";
+
+const MSSQL_PORT = 1433;
+
+type ProductKey = `${string}-${string}-${string}-${string}-${string}`;
+
+type Edition = "Evaluation" | "Developer" | "Express" | "Web" | "Standard" | "Enterprise" | ProductKey;
+
+const PASSWORD_CATEGORY_VALIDATION_PATTERNS = [/[A-Z]+/, /[a-z]+/, /[0-9]+/, /[@$!%*?&]+/];
+
+export class MSSQLServerContainer extends GenericContainer {
+  private database = "master";
+  private username = "SA";
+  private password = "Passw0rd";
+  private edition: Edition = "Developer";
+  private acceptEula = "N";
+
+  constructor(image = "mcr.microsoft.com/mssql/server:2022-latest") {
+    super(image);
+  }
+
+  public acceptLicense(): this {
+    this.acceptEula = "Y";
+    return this;
+  }
+
+  public withDatabase(database: string): this {
+    this.database = database;
+    return this;
+  }
+
+  public withPassword(password: string): this {
+    this.checkPasswordStrength(password);
+    this.password = password;
+    return this;
+  }
+
+  public withEdition(edition: Edition): this {
+    this.edition = edition;
+    return this;
+  }
+
+  private checkPasswordStrength(password: string) {
```

Comment:
For my own understanding, why is it necessary to enforce a strong password here?

Reply:
SQL Server requires a password to meet the following criteria:

The password doesn't contain the account name of the user.

The password is at least eight characters long.

The password contains characters from three of the following four categories:

1. Latin uppercase letters (A through Z)
2. Latin lowercase letters (a through z)
3. Base 10 digits (0 through 9)
4. Non-alphanumeric characters such as: exclamation point (!), dollar sign ($), number sign (#), or percent (%).

[source](https://learn.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver16)

This implementation comes from Java but as @eddumelendez stated above, it looks like they're changing it so that the service will handle it.